components/Controllers/AuctionPageContent/AuctionPageContent.tsx

```diff
@@ -45,38 +43,70 @@ export function AuctionPageContent({
     hourlyPriceChange,
     priceUpdated,
   } = useLiveAuctionPrice(auction, 8000);
+  const topBid = useAuctionTopBid(auction);
 
   const [timeElapsed, setTimeElapsed] = useState<number>(
     currentTimeInSeconds() - auction.start.timestamp,
   );
 
-  const ethPrice = useAsyncValue(() => {
+  const auctionCompleted = useMemo(() => {
+    return !!auction.end;
+  }, [auction.end]);
+
+  const auctionUnderlyingDisplayPrice = useMemo(() => {
+    if (!auctionCompleted) {
+      if (!liveAuctionPriceUnderlying) return null;
+      else
+        return formatBigNum(
+          liveAuctionPriceUnderlying,
+          controller.underlying.decimals,
+        );
+    } else {
+      if (!paprPrice) return null;
+      const auctionPriceNumber = parseFloat(
+        ethers.utils.formatUnits(
+          liveAuctionPrice,
+          controller.paprToken.decimals,
+        ),
+      );
+      return (auctionPriceNumber * paprPrice).toFixed(4);
+    }
+  }, [
+    auctionCompleted,
+    liveAuctionPriceUnderlying,
+    liveAuctionPrice,
+    paprPrice,
+    controller.underlying.decimals,
+    controller.paprToken.decimals,
+  ]);
+
+  const ethToUSDPrice = useAsyncValue(() => {
     return getUnitPriceForEth(
       'usd',
-      configs[tokenName as SupportedToken].network as SupportedNetwork,
+      configs[config.tokenName as SupportedToken].network as SupportedNetwork,
```

Comment:
why are we looking up a config based on the token name we pulled from the config?

Reply:
lol, good call, simplified